test/util.js

```diff
@@ -0,0 +1,15 @@
+var path = require('path');
+var fs = require('fs');
+
+
+// ensure directory exists, from
+// http://stackoverflow.com/a/34509653/2039
+var ensureDirectoryExistence = function(filePath) {
+  var dirname = path.dirname(filePath);
+  if (fs.existsSync(dirname)) {
```

Comment:
is there a reason you're `fs.existsSync()` here and `fs.statSync()` in index.js? I wasn't sure of the best way myself, came across this http://stackoverflow.com/a/4482701/656010 which lead me to the [fs docs which confirm that it's deprecated](https://nodejs.org/api/fs.html#fs_fs_existssync_path). Unless there's a good reason, probably should stick w/ `fs.statSync()`